### PR TITLE
[KHRGA-76] Updated host accessors in Data access

### PR DESCRIPTION
**Pull request by imvucic**
Issue: KHRGA-76

### CODE REVIEW

File: source/iface/host_accessor.rst [hunk 1]
```diff
@@ -2,8 +2,656 @@
...
-***********************
+=======================
+
+::
+
+  class host_accessor;
+
+The ``host_accessor`` class provides access to data in a ``buffer``
+from host code that is outside of a command (i.e. do not use this
+class to access a buffer inside a host task).
```
Comment (mkinsner) [inline source/iface/host_accessor.rst:19]:
```suggestion
class to access a buffer from inside device code or a host task).
```

File: source/iface/host_accessor.rst [hunk 2]
```diff
@@ -2,8 +2,656 @@
...
+As with ``accessor``, the dimensionality of ``host_accessor``
+must match the underlying buffer, however, there is a special case
+if the buffer is one-dimensional. In this case, the accessor may
+either be one-dimensional or it may be zero-dimensional.
+A zero-dimensional accessor has access to just the first element of
+the buffer, whereas a one-dimensional accessor
+has access to the entire buffer.
+
+The ``host_accessor`` class supports the following access modes:
+``access_mode::read``,`` access_mode::write``
```
Comment (mkinsner) [inline source/iface/host_accessor.rst:30]:
```suggestion
``access_mode::read``,``access_mode::write``
```

Comment (mkinsner) [reply source/iface/host_accessor.rst:30]: This fix needed for the text to render as code

File: source/iface/host_accessor.rst [hunk 3]
```diff
@@ -2,8 +2,656 @@
...
+  :header-rows: 1
+
+  * - Data type
+    - Access mode
+  * - ``not const-qualified``
+    - ``access_mode::read``
+  * - ``const-qualified``
+    - ``access_mode::read``
+
+The specializations of ``host_accessor``
```
Comment (mkinsner) [inline source/iface/host_accessor.rst:167]:
```suggestion
These specializations of ``host_accessor``
```

File: source/iface/host_accessor.rst [hunk 4]
```diff
@@ -2,8 +2,656 @@
...
+that are read-only accessors.
```
Comment (mkinsner) [inline source/iface/host_accessor.rst:168]:
```suggestion
are read-only accessors.
```

File: source/iface/host_accessor.rst [hunk 5]
```diff
@@ -2,8 +2,656 @@
...
+.. list-table::
+  :header-rows: 1
+
+  * - Data type
+    - Access mode
+  * - ``not const-qualified``
+    - ``access_mode::read_write``
+
+The table present an implicit conversion from
+the read-write specialization.
```
Comment (mkinsner) [inline source/iface/host_accessor.rst:184]: I don't think the table presents that, so this wording is not correct.  The spec text following relates two tables, which is different from the wording here proposed for the reference.

> There is also an implicit conversion from the read-write host_accessor type shown in [Table 75](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.accessors.host.buffer.read-write) to any of the read-only accessors in [Table 74](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#table.accessors.host.buffer.read-only), provided that all other template parameters are the same.

File: source/iface/host_accessor.rst (hunk 1 above)
Comment (imvucic) [reply source/iface/host_accessor.rst]: Fixed wit commit : https://github.com/KhronosGroup/SYCL_Reference/pull/98/commits/0a37df862ec9016a9e55cb029c008f842e1d09ab

File: source/iface/host_accessor.rst (hunk 2 above)
Comment (imvucic) [reply source/iface/host_accessor.rst]: Fixed wit commit : https://github.com/KhronosGroup/SYCL_Reference/pull/98/commits/0a37df862ec9016a9e55cb029c008f842e1d09ab

File: source/iface/host_accessor.rst (hunk 3 above)
Comment (imvucic) [reply source/iface/host_accessor.rst]: Fixed wit commit : https://github.com/KhronosGroup/SYCL_Reference/pull/98/commits/0a37df862ec9016a9e55cb029c008f842e1d09ab

File: source/iface/host_accessor.rst (hunk 4 above)
Comment (imvucic) [reply source/iface/host_accessor.rst]: Fixed wit commit : https://github.com/KhronosGroup/SYCL_Reference/pull/98/commits/0a37df862ec9016a9e55cb029c008f842e1d09ab

File: source/iface/host_accessor.rst (hunk 5 above)
Comment (imvucic) [reply source/iface/host_accessor.rst]: Please check if this context present correct state: https://github.com/KhronosGroup/SYCL_Reference/pull/98/commits/0a37df862ec9016a9e55cb029c008f842e1d09ab

**Review by mkinsner — APPROVED** [review]

**Pull request merged**